### ISSUE TITLE
Ruby matching with for in method names

### ISSUE THREAD

**Issue by woutersmeenk** [comby-tools/comby#330]
**Describe the bug**
When matching ruby code expression names that contain `_for` are valid matches to not match

**Reproducing**
bit.ly/3Hezuwj

When `bla_for` is replaced with `blafor` the above example does match

**Expected behavior**
I expected the above example (without modification) to match.

**Additional context**
It seems the `_for` in the expression name is being parsed as a delimiter

**Comment by rvantonder** [comment]
yep, this is wrong behavior. thanks for the report, will have a look at a fix!

**Comment by schreiaj** [comment]
Also seeing similar behavior with `do_`

bit.ly/3diyFsl

**Comment by oskarkook** [comment]
Same in Elixir with `end` for example: bit.ly/3NE5nSR
